scipy/stats/tests/test_stats.py

```diff
@@ -1402,6 +1402,21 @@ def test_mode_nan(self):
         assert_raises(ValueError, stats.mode, data1, nan_policy='raise')
         assert_raises(ValueError, stats.mode, data1, nan_policy='foobar')
 
+    def test_smallest_equal(self):
```

Comment:
Some concision might be gained by writing a [parametrized test](https://docs.pytest.org/en/latest/parametrize.html) here, with i.e., a "data" parameter which corresponds to each of the data lists, and then a single result & assert statement. One possible drawback is that you'd also likely need another parameter for `nan_policy` specification, and could end up running a few more test permutations by default, but that might be ok.